### DISCUSSION TITLE
Danfoss Ally with external Temperature Sensor

### DISCUSSION THREAD

**Discussion by 735751363r** [Koenkk/zigbee2mqtt#11257] [Devices]
Hi,
first of all, I`m totally new to HomeAssistant, Zigbee and MQTT.
I bought a pair of Danfoss Ally  Zigbee TRV`s with the intention to use them with an external Tuya TS0201 Thermometer and Humidity Sensor. The TRV seems to have the option of using external sensors through the "radiator_covered" and "external_measured_room_sensor" option. 
![TRV](https://user-images.githubusercontent.com/59930504/152680850-7aba1095-b2c4-4592-9009-8887e85ef7f0.JPG)

My big and eventually stupid/noobish question is: Ho do I get this set up? How do I get the TRV linked to the external Sensor? Where do I enter the entetie or IEEE number for the TRV to access the Thermometer

Maybe someone can explain it to me on an beginer level.
Thanks
Julius

**Comment by Mikej0h** [comment]
Hi Julius,

You don't actually technically link the devices together.

You publish a message to your TRV via a trigger (or some sorts) via Home Assistant to Z2M, and sort of 'calibrate' your TRV with the value of the TS0201. 
So you read the current temperature of the sensor and publish as the 'external_measured_room_sensor' value to the TRV.

KR, Mike

**Reply by 735751363r** [reply]
Thank you Mike for this Hint and for your reply,
maybe it is obvious for HA MQTT or Z2M experienced but for me it is not. Like I understand your answer, I have to write an automation that sends a Message to MQTT to the TVR if the temperature changes?!
I tried the following:
![trigger](https://user-images.githubusercontent.com/59930504/152999532-21356bc2-de7d-431a-83dc-7f885cbb56d1.JPG)
I made this trigger in Automations if the Measured temperature is belo or over a specific Value, that leads to this  action:
![action](https://user-images.githubusercontent.com/59930504/152999784-cf30e53c-923d-47e1-a006-a591301fb859.JPG)
the marked Digits are the IEEE Adress of the trv. I found these formulations/code on this website:
https://www.zigbee2mqtt.io/devices/014G2461.html
am I on the right track or completely wrong?
Thank you!
Julius

**Reply by Mikej0h** [reply]
Hi Julius,

Lets first start with the easy bit - your device names.
When you initially pair these, they have a randomized number, together with your IEEE address - however, after pairing you can set an alias in the GUI of Z2M. You can find this in the "Devices" tab, on the right side of each device click the blue icon. Make sure to also enable the toggle for "Update Home Assistant entity ID". After you have done so, you can use the chosen "friendly" names.

There are 100 ways to Rome to get your integration working. Please note that what you want to-do is not an easy one, as I don't think the UI directly supports it, so you must write something in YAML.
I suggest you to look into Blueprints that other community members already wrote.
One that seem to-do what you want is https://community.home-assistant.io/t/sync-trv-with-external-tempature-sensor/298024
While mostly fitting, your Danfoss requires the temperature in 100's, so when the temperature is 21c, the correction temperature must be 2100.

Hopefully you understand me when I say that you don't really have an issue or question of Z2M, but mostly on generally automating in HA. The support forums of HA would be your best bet.

Greetings, Mike

**Reply by 735751363r** [reply]
Thank you Mike for your further support! I have been away last week and will try it the next days Thank's so far!
